lib/jekyll/assets/proxies/magick.rb

```diff
@@ -9,10 +9,9 @@
   presets = %W(@2x @4x @1/2 @1/3 @2/3 @1/4 @2/4 @3/4
     @double @quadruple @half @one-third @two-thirds @one-fourth
       @two-fourths @three-fourths)
+  ARGS=args.freeze
 
-  Jekyll::Assets::Env.liquid_proxies.add :magick, :img, *(args + presets) do
-    PRESETS = presets
-    ARGS = args
```

Comment:
So it seems, I guess I am crazy for not noticing, any way we can just drop the local variables and switch entirely to a frozen array from the start?  I think that would solve all confusion in the future.